battleground/persistence/agent_data.py

```diff
@@ -17,17 +17,22 @@ def insert_new_agent(owner, name, game_type, db_handle):
     return agent_id
 
 
-def get_agent_id(owner, name, game_type, db_handle=None):
+def get_agent_id(owner, name, game_type, db_handle=None, taken_ids=None):
```

Comment:
My guess is that this is still not working as expected, I will fix this on the next iteration.